### PR TITLE
Remove creation of APM provileges.

### PR DESCRIPTION
**Pull request by jalvz**
As per https://github.com/elastic/apm-server/issues/2987#issuecomment-577717543

### CODE REVIEW

File: cmd/apikey.go [hunk 1]
```diff
@@ -272,20 +269,6 @@ func booleansToPrivileges(ingest, sourcemap, agentConfig bool) []es.PrivilegeAct
 // we need to ensure forward-compatibility, for which future privileges must be created here and
 // during server startup because we don't know if customers will run this command
 func createAPIKeyWithPrivileges(client es.Client, keyName, expiry string, privileges []es.PrivilegeAction, asJSON bool) error {
```
Comment (simitt) [inline cmd/apikey.go]: Can you rename to `createAPIKey` please.

File: elasticsearch/security_api.go [hunk 2]
```diff
@@ -104,10 +85,6 @@ type GetAPIKeyResponse struct {
 	APIKeys []APIKeyResponse `json:"api_keys"`
 }
 
-type CreatePrivilegesRequest map[AppName]PrivilegeGroup
-
-type CreatePrivilegesResponse map[AppName]PrivilegeResponse
```
Comment (simitt) [inline elasticsearch/security_api.go]: I think [`PrivilegeResponse` and `PrivilegeGroup`](https://github.com/elastic/apm-server/pull/3251/files#diff-39e6b7ff4f3bd5569c98e478c2740cdbR138-R140) is also unused now

Comment (jalvz) [reply elasticsearch/security_api.go]: good catch, somehow my IDE didn't flag it

**Comment by axw** [comment]
Verified with BC3:

<details>
<summary>No application privileges created by `apikey create`</summary>

```
$ apm-server apikey create -E output.elasticsearch.username=admin -E output.elasticsearch.password=changeme
API Key created:

Name ........... apm-key
Expiration ..... never
Id ............. IgVl9G8BagmrQdVeveZQ
API Key ........ QfaT6SdaTb-zELUw9IpgeQ (won't be shown again)
Credentials .... SWdWbDlHOEJhZ21yUWRWZXZlWlE6UWZhVDZTZGFUYi16RUxVdzlJcGdlUQ== (use it as "Authorization: APIKey <credentials>" header to communicate with APM Server, won't be shown again)

$ apm-server apikey verify --credentials SWdWbDlHOEJhZ21yUWRWZXZlWlE6UWZhVDZTZGFUYi16RUxVdzlJcGdlUQ==
Authorized for privilege "config_agent:read"...: Yes
Authorized for privilege "event:write"...: Yes
Authorized for privilege "sourcemap:write"...: Yes

$ curl http://admin:changeme@localhost:9200/_security/privilege/apm
{}
```

</details>

<details>
<summary>User without privileges receives error when trying to create API key</summary>

```
$ curl http://admin:changeme@localhost:9200/_security/role/api_key_creator?pretty
{
  "api_key_creator" : {
    "cluster" : [
      "manage_own_api_key"
    ],
    "indices" : [ ],
    "applications" : [ ],
    "run_as" : [ ],
    "metadata" : { },
    "transient_metadata" : {
      "enabled" : true
    }
  }
}

$ apm-server apikey create -E output.elasticsearch.username=api_key_creator -E output.elasticsearch.password=changeme
api_key_creator is missing the following requested privilege(s): sourcemap:write, event:write, config_agent:read.

You might try with the superuser, or add the APM application privileges to the role of the authenticated user, eg.:
PUT /_security/role/my_role {
        ...
        "applications": [{
          "application": "apm",
          "privileges": ["sourcemap:write", "event:write", "config_agent:read"],
          "resources": ["*"]
        }],
        ...
}
```

</details>

<details>
<summary>User with privileges can successfully create an API key</summary>

```
$ curl http://admin:changeme@localhost:9200/_security/role/api_key_creator?pretty
{
  "api_key_creator" : {
    "cluster" : [
      "manage_own_api_key"
    ],
    "indices" : [ ],
    "applications" : [
      {
        "application" : "apm",
        "privileges" : [
          "sourcemap:write",
          "event:write",
          "config_agent:read"
        ],
        "resources" : [
          "*"
        ]
      }
    ],
    "run_as" : [ ],
    "metadata" : { },
    "transient_metadata" : {
      "enabled" : true
    }
  }
}

$ apm-server apikey create -E output.elasticsearch.username=api_key_creator -E output.elasticsearch.password=changeme
API Key created:

Name ........... apm-key
Expiration ..... never
Id ............. KwZ29G8BagmrQdVeRBYr
API Key ........ 2QvEiThRRmScLy6wPRa2pQ (won't be shown again)
Credentials .... S3daMjlHOEJhZ21yUWRWZVJCWXI6MlF2RWlUaFJSbVNjTHk2d1BSYTJwUQ== (use it as "Authorization: APIKey <credentials>" header to communicate with APM Server, won't be shown again)

$ apm-server apikey verify --credentials S3daMjlHOEJhZ21yUWRWZVJCWXI6MlF2RWlUaFJSbVNjTHk2d1BSYTJwUQ==
Authorized for privilege "config_agent:read"...: Yes
Authorized for privilege "event:write"...: Yes
Authorized for privilege "sourcemap:write"...: Yes
```

</details>

<details>
<summary>No more `--purge` flag on `apikey invalidate`</summary>

```
$ apm-server apikey invalidate --purge
Error: unknown flag: --purge
Usage:
  apm-server apikey invalidate [flags]

Flags:
      --id string     id of the API Key to delete
      --name string   name of the API Key(s) to delete (several might match)
      --json          prints the output of this command as JSON
  -h, --help          help for invalidate

Global Flags:
  -E, --E setting=value      Configuration overwrite
  -c, --c string             Configuration file, relative to path.config (default "apm-server.yml")
  -d, --d string             Enable certain debug selectors
  -e, --e                    Log to stderr and disable syslog/file output
      --path.config string   Configuration path
      --path.data string     Data path
      --path.home string     Home path
      --path.logs string     Logs path
      --plugin pluginList    Load additional plugins
      --strict.perms         Strict permission checking on config files (default true)
  -v, --v                    Log at INFO level
```

</details>

**Pull request merged**